src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -263,6 +309,7 @@ public AnomalyDetector(StreamInput input) throws IOException {
             user = null;
         }
         detectorType = input.readOptionalString();
+        mlModelId = input.readOptionalString();
```

Comment:
Suggest to put new fields to the end of this method for backward compatibility. In old AD code, we don't have this field, the output stream will not have this optional string, so the input stream may fail to read.
Change to this and add to last line, similar for `writeTo`
```
if (input.available() > 0) {
    mlModelId = input.readOptionalString();
}
```

Reply:
Definitely, thanks! For `writeTo`, I put `output.writeOptionalString(mlModelId);` at last line.